From a43f5002cdaad6bc00ba36c386ca6eec3def4ff5 Mon Sep 17 00:00:00 2001
From: "Huang, Tao" <huangtao@rock-chips.com>
Date: Tue, 29 Mar 2016 19:40:03 +0800
Subject: [PATCH] Revert "mmc: block: ioctl return EINVAL if cmd unknown"

This reverts commit 7c1abbfe2a27bae41202d4b46687f82eba0c1d5b.
Will fixes by aosp commit "mmc: block: fix ABI regression of mmc_blk_ioctl"

Signed-off-by: Huang, Tao <huangtao@rock-chips.com>
---
 drivers/mmc/card/block.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index 6efc81ea5dc5..7befdaee555a 100644
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -690,11 +690,6 @@ static int mmc_blk_ioctl_multi_cmd(struct block_device *bdev,
 static int mmc_blk_ioctl(struct block_device *bdev, fmode_t mode,
 	unsigned int cmd, unsigned long arg)
 {
-#ifdef CONFIG_ARCH_ROCKCHIP
-	if (cmd != MMC_IOC_CMD && cmd != MMC_IOC_MULTI_CMD)
-		return -EINVAL;
-#endif
-
 	/*
 	 * The caller must have CAP_SYS_RAWIO, and must be calling this on the
 	 * whole block device, not on a partition.  This prevents overspray
-- 
2.35.3

